pkg/depth/buffer.go

```diff
@@ -230,29 +259,43 @@ func (b *Buffer) fetchAndPush() error {
 	}
 
 	var pushUpdates []Update
-	for idx, u := range b.buffer {
-		// skip old events
-		if u.FinalUpdateID <= finalUpdateID {
-			continue
+	if b.isFutures {
+		for _, u := range b.buffer {
+			if u.FinalUpdateID < finalUpdateID {
+				continue
+			}
+			if u.FirstUpdateID <= finalUpdateID && u.FinalUpdateID >= finalUpdateID {
+				pushUpdates = append(pushUpdates, u)
+			}
```

Comment:
I think this logic is the same as non-futures? it's just reversed ?

Reply:
And the field `PreviousUpdateID` is not used here?

Reply:
I see, it s indeed the same logic. I ll merge it.